assets/js/components/HypothesisOpener.js

```diff
@@ -330,6 +351,39 @@ module.exports = class HypothesisOpener {
     return positioners[articleType] || positioners.default;
   }
 
+  static findTarget($contentContainer) {
+    let target = null;
+    const centre = Math.floor(($contentContainer.querySelectorAll('p').length - 1) / 2);
+    [].forEach.call($contentContainer.querySelectorAll('p'), (possible, index) => {
```

Comment:
Not that i'd worry about this now, but this could be not improved so to say but neatened up...

```
target = $contentContainer.querySelectorAll('p').find( (possible) => {
  if (
        target === null &&
        index >= centre &&
        !HypothesisOpener.targetWithinInLineProfile(possible) &&
        !HypothesisOpener.targetWithinBlockquote(possible) &&
        !HypothesisOpener.targetBelowSectionHeading(possible)
      ) {
        return true;
      }
  return false;
});
```

Reply:
I tried this but it doesn't seem to work on the output of `$contentContainer.querySelectorAll('p').find`. I think I would need to convert `$contentContainer.querySelectorAll('p')` to an array and this would risk losing the context of the elements. Since we need to interrogate the parentNodes of these elements we can't afford to lose the context. I wish this would work as it is much cleaner than my code.

Reply:
Yep, my bad. Seems it returns a [nodelist](https://developer.mozilla.org/en-US/docs/Web/API/NodeList) rather than an array and hence the filter isn't valid. Standard loop would work though...
```
for (const possible in $contentContainer.querySelectorAll('p')) {
    if ( index >= centre &&
        !HypothesisOpener.targetWithinInLineProfile(possible) &&
        !HypothesisOpener.targetWithinBlockquote(possible) &&
        !HypothesisOpener.targetBelowSectionHeading(possible)
      ) {
        target = possible;
        break;
      }
}
```

Reply:
Honestly though, just commit it as is.